test/rhsmlib/base.py

```diff
@@ -68,210 +64,57 @@ def injection_definitions(self, *args, **kwargs):
-    blocks the main thread (the DBus call must be asynchronous to avoid deadlock).  The function passed to
-    dbus_request is run and then dbus_request unblocks the main thread.
-    """
-
-    def setUp(self):
-        super(DBusObjectTest, self).setUp()
-        self.started_event = threading.Event()
-        self.stopped_event = threading.Event()
-        self.handler_complete_event = threading.Event()
-        # This is up a level because it needs to be defined before the server is instantiated
-        self.mock_identity = mock.Mock(spec=Identity, name="Identity").return_value
-        self.mock_identity.cert_dir_path = (
-            "path.txt"  # must be a string, otherwise it is set as a mock object
-        )
-        # and os.path throws type error, causing tests to hang
+class DBusServerStubProvider(unittest.TestCase):
+    """Special class used start a DBus server.
 
-        sender_patcher = mock.patch("rhsmlib.client_info.DBusSender.get_cmd_line")
-        self.mock_sender_get_cmd_line = sender_patcher.start()
-        self.mock_sender_get_cmd_line.return_value = "unit-test"
-        self.addCleanup(sender_patcher.stop)
+    All rhsmlib.objects.*.*DbusObject classes need a connection, object path
+    and a bus name to be instantiated. The functions they expose over DBus API
+    are converted to via decorators to special methods the dbus-python library
+    can use, but we can use `__wrapped__` attribute of these methods to obtain
+    original Python functions.
 
-        # If we don't use a BusConnection and use say dbus.SessionBus() directly, each test will end up
-        # getting old bus connections since the dbus bindings cache bus connections.  You can use the private
-        # kwarg to tell dbus.SessionBus/SystemBus not to cache, but that's deprecated.
-        self.server_thread = ServerThread(
-            kwargs={
-                "object_classes": self.dbus_objects(),
-                "bus_name": self.bus_name(),
-                "bus_class": dbus.bus.BusConnection,
-                "bus_kwargs": self.bus_kwargs,
-                "started_event": self.started_event,
-                "stopped_event": self.stopped_event,
-            }
-        )
-        self.started_event.wait()
-        self.result_queue = queue.Queue(maxsize=1)
-        self.addCleanup(self.stop_server)
+    This will allow us to test just our implementation, without full
+    communication over DBus.
+    """
 
-    def stop_server(self):
-        self.server_thread.stop()
-        self.stopped_event.wait()
+    bus: dbus.service.BusName = None
+    """Connection object for the DBus RHSM API classes."""
 
-    @property
-    def bus_kwargs(self):
-        if os.geteuid() == 0:
-            return {"address_or_type": dbus.Bus.TYPE_SYSTEM}
-        else:
-            return {"address_or_type": dbus.Bus.TYPE_SESSION}
+    dbus_class: type = NotImplemented
+    """DBus RHSM API class, subclass of `BaseObject`."""
```

Comment:
Can you please empty lines between each class attributes. It is little bit confusing, when it is all together with doc strings.